interfaces/inc/c_pal/malloc_multi_flex.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#ifndef MALLOC_MULTI_FLEX_H
+#define MALLOC_MULTI_FLEX_H
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    #define ARG_OVERFLOW_CHECK(arg) \
+        if ((MU_C2(arg, _size) != 0 && SIZE_MAX / MU_C2(arg, _size) < MU_C2(arg, _count)) || (SIZE_MAX - size_tracker < MU_C2(arg, _count) * MU_C2(arg, _size)))\
+        {\
+            return NULL;\
+        }\
+        size_tracker += MU_C2(arg, _count) * MU_C2(arg, _size);
+
+    #define ARGS_OVERFLOW_CHECK(...) \
+        size_t size_tracker = parent_struct_size;\
+        MU_FOR_EACH_1(ARG_OVERFLOW_CHECK, __VA_ARGS__)
+
+    #define ARG_LIST_VALUE_ARG(arg) \
+        , uint32_t MU_C2(arg, _count), uint32_t MU_C2(arg, _size)
+
+    #define ARG_LIST_VALUES(...) \
+        MU_FOR_EACH_1(ARG_LIST_VALUE_ARG, __VA_ARGS__)
+
+    #define ASSIGN_INTERNAL_STRUCT_PTR(arg) \
+        parent_struct_pointer->arg = pointer_iterator;\
+        pointer_iterator = (char*)pointer_iterator + MU_C2(arg, _count) * MU_C2(arg, _size);;
+
+    #define ASSIGN_INTERNAL_STRUCT_PTRS(...) \
+        MU_FOR_EACH_1(ASSIGN_INTERNAL_STRUCT_PTR, __VA_ARGS__)
+
+    //Codes_MALLOC_MULTI_FLEX_24_005: [ MALLOC_MULTI_FLEX shall expand type to the name of the malloc function in the format of: MALLOC_MULTI_FLEX_type. ]
+    #define MALLOC_MULTI_FLEX(type) \
+        MU_C2(malloc_multi_flex, type) \
+
+    #define DEFINE_MALLOC_MULTI_FLEX(type, ...)\
```

Comment:
>DEFINE_MALLOC_MULTI_FLEX [](http://example.com/codeflow?start=12&length=24)

I think this should have a DECLARE/DEFINE pair like any other template that we build.

Reply:
declare would be used in a header, define would be used in .c.
That gives the flexibility to the user to split the declaration/definition in .h and .c or to keep them both in the same file.